src/models.py

```diff
@@ -496,52 +527,60 @@ def build_reddit_module(task, d_inp, params):
 
 
 def build_image_sent_module(task, d_inp, params):
-    pooler = Pooler.from_params(d_inp, params['d_proj'])
+    pooler = Pooler(d_inp, d_proj=params['d_proj'])
     return pooler
 
 
-def build_single_sentence_module(task, d_inp, params):
+def build_single_sentence_module(task, d_inp, model, params):
     ''' Build a single classifier '''
-    pooler = Pooler.from_params(d_inp, params['d_proj'])
-    classifier = Classifier.from_params(params['d_proj'], task.n_classes, params)
+    pool_type = "first" if model.use_bert else "max"
+    pooler = Pooler(d_inp, project=not model.use_bert, d_proj=params['d_proj'], pool_type=pool_type)
+    d_out = d_inp if model.use_bert else params["d_proj"]
+    classifier = Classifier.from_params(d_out, task.n_classes, params)
     return SingleClassifier(pooler, classifier)
 
 
-def build_pair_sentence_module(task, d_inp, model, vocab, params):
+def build_pair_sentence_module(task, d_inp, model, params):
```

Comment:
Same as above, avoid passing large objects like `model` around if we only need a couple of fields.